bin/autodeploy.sh

```diff
@@ -0,0 +1,416 @@
+  msg "INFO: This will setup up the Hetzner system with basic defaults, download and unpack the wire-server-deploy artifact."
+  sleep 5
+  export LC_ALL="C.UTF-8";
+  ansible-playbook ../ansible/hetzner-single-deploy.yml -e "artifact_hash=$ARTIFACT_HASH" -i $SSH_USER@webapp."$TARGET_SYSTEM", --diff 
+}
+
+remote_deployment() {
+  msg() {
+    echo >&2 -e "${1-}"
+  }
+  cd $SCRIPT_DIR &>/dev/null || exit 1
+
+  bash bin/offline-vm-setup.sh
+  msg ""
+  while sudo virsh list --all | grep -Fq running; do
+    sleep 20
+    msg "INFO: VM deployment still in progress ..."
+  done
+  sleep 20
+  msg ""
+  msg "INFO: VM deployment done. Starting all VMs:"
+  msg ""
+  for VM in $(sudo virsh list --all --name); do sudo virsh start "$VM"; done
+  sleep 60
+
+  msg ""
+  msg "INFO: Setting up offline environment (this will take a while)."
+  msg ""
+  # Rather than sourcing wire-server-deploy/bin/offline-env.sh, we invoke
+  # the relevant commands below, declaring "d" as a function instead of an alias.
+  ZAUTH_CONTAINER=$(sudo docker load -i "$SCRIPT_DIR"/containers-adminhost/quay.io_wire_zauth_*.tar | awk '{print $3}')
+  export ZAUTH_CONTAINER
+  WSD_CONTAINER=$(sudo docker load -i "$SCRIPT_DIR"/containers-adminhost/container-wire-server-deploy.tgz | awk '{print $3}')
+  d() {
+    sudo docker run --network=host -v "${SSH_AUTH_SOCK:-nonexistent}":/ssh-agent -e SSH_AUTH_SOCK=/ssh-agent -v "$HOME"/.ssh:/root/.ssh -v "$PWD":/wire-server-deploy "$WSD_CONTAINER" "$@"
+  }
+  export -f d
+
+  bash bin/offline-secrets.sh
+
+  HOST_IP=$(dig @resolver4.opendns.com myip.opendns.com +short)
+
+  cat >ansible/inventory/offline/hosts.ini<<EOF
+[all]
+assethost ansible_host=192.168.122.10
+kubenode1 ansible_host=192.168.122.21
+kubenode2 ansible_host=192.168.122.22
+kubenode3 ansible_host=192.168.122.23
+ansnode1 ansible_host=192.168.122.31
+ansnode2 ansible_host=192.168.122.32
+ansnode3 ansible_host=192.168.122.33
+
+[all:vars]
+ansible_user = $DEMO_USER
+
+[cassandra:vars]
+cassandra_network_interface = enp1s0
+cassandra_backup_enabled = False
+cassandra_incremental_backup_enabled = False
+
+[elasticsearch:vars]
+elasticsearch_network_interface = enp1s0
+
+[minio:vars]
+minio_network_interface = enp1s0
+prefix = ""
+domain = "$TARGET_SYSTEM"
+deeplink_title = "wire demo environment, $TARGET_SYSTEM"
+
+[rmq-cluster:vars]
+rabbitmq_network_interface = enp1s0
+
+[kube-master]
+kubenode1
+kubenode2
+kubenode3
+
+[etcd]
+kubenode1 etcd_member_name=etcd1
+kubenode2 etcd_member_name=etcd2
+kubenode3 etcd_member_name=etcd3
+
+[kube-node]
+kubenode1
+kubenode2
+kubenode3
+
+[k8s-cluster:children]
+kube-master
+kube-node
+
+[cassandra]
+ansnode1
+ansnode2
+ansnode3
+
+[cassandra_seed]
+ansnode1
+
+[elasticsearch]
+ansnode1
+ansnode2
+ansnode3
+
+[elasticsearch_master:children]
+elasticsearch
+
+[minio]
+ansnode1
+ansnode2
+ansnode3
+
+[rmq-cluster]
+ansnode1
+ansnode2
+ansnode3
+EOF
+
+  d ./bin/offline-cluster.sh
```

Comment:
possibly dumb question, but where does `d` come from? It's hard to find a single-letter symbol.